src/core/rules.js

```diff
@@ -490,9 +490,69 @@ class Rule {
 Rule.logger = new Logger('Rule');
 
 class Rules extends Rule {
-    constructor() {
+    constructor(excludeOn = null) {
         super();
         this.rules = [];
+        this.excludeOn = null;
+
+        if (excludeOn !== null) {
+            if (!Array.isArray(excludeOn)) {
+                throw new Error(`excludeOn parameter of Rules must be either null or an Array of strings`);
+            }
+
+            if (excludeOn.length > 0) {
+                this.excludeOn = excludeOn;
+            }
+        }
+
+        this.init();
+    }
+
+    init() {
+        if (this.excludeOn === null) {
+            return;
+        }
+
+        // Stores the processed excludeOn entries
+        this.__excludeOn = [];
+
+        // eslint-disable-next-line no-useless-escape
+        const regex = new RegExp(/^([^\.]+)\.([^\.]+)(?:\.([^\.]+))?$/, 'm');
+
+        for (let typeFeatureSelector of this.excludeOn) {
+            const matches = typeFeatureSelector.match(regex);
+
+            regex.lastIndex = 0;
+
+            if (matches === null) {
+                throw new Error(`Invalid typeFeature selector "${typeFeatureSelector}" found in excludeOn constraints in given Rules`);
+            }
+
+            const entry = {
+                type: matches[1],
```

Comment:
matches[<number>] (Magic numbers). Constants could be used

Reply:
Since it's used at this location only, the meaning of the matching groups directly correlates with the fields in the entry object. It's a bit hard to have constants as names here, because the meaning actually changes from types to talent namespaces once the third group is matched.

Reply:
I agree.